docs/rfc/active/000-internationalization.md

```diff
@@ -0,0 +1,165 @@
+- Start Date: (2022-09-14)
+- RFC PR: (after opening the RFC PR, update this with a link to it and update the file name)
+- Discussion Issue: (None)
+- Implementation PR(s): (leave this empty)
+
+# Internalization (i18n)
+
+## Summary
+> Implements the internalization of Datahub.
+> Being able to change and add language without modifying the source.
+
+## Basic example
+```javascript
+// Instead of this
+<Button type="submit">
+    Next
+</Button>
+// We write this
+<Button type="secondary">
+    {t('common.next')}
+</Button>
+// Or
+<Button type="secondary">
+    <Trans i18nKey="common.next"/>
+</Button>
+// And the display is the same in english, but translated in choosen language.
+```
+
+## Motivation
+Why are we doing this?
+> I have implemented for my company a french version of Datahub (or 90% of it). 
+> Now I have difficulties pulling new versions so I think it would be beneficial for both of 
+> us to share my modifications.
+>
+What use cases does it suppor
+> Anyone who would want to have their own language for the Datahub will be able to do so.
+>
+What is the expected outcome?
+> If I finish this as I expect it, Datahub will be able to change dynamically his language, no reload needed.
+> And anybody will be able to add a language just by adding a *translation.json* file in the correct directory.
+>
+
+## Requirements
+> - [ ] react-i18next
+> - [x] Uniformisation of Datahub types
```

Comment:
Anything that is based on server driven text will indeed be a challenge. The most notable examples that come to mind:

- Error messages from the GraphQL api
- Error messages and logs that come from the ingestion runner 
- Tutorial content (educational content) - This actually does not come from server but still will need to be considered. Examples are the welcome page on initial deployment along with a new new-user onboarding experience we're currently working on

Reply:
From my experience about error message, we can add to any error thrown a new "translateKey" field and on the front we catch any error from serveur and apply the translation to the key.

No deep changes since we keep the existing text from exceptions, Logs will still works the same with only english messages.

Reply:
For the tutorial content, I was not worried about it but I haven't dig into it. I'll try to make a closer look to understand what is the complexity you are talking about.